src/main/java/hudson/plugins/octopusdeploy/OctoInstallation.java

```diff
@@ -141,13 +141,15 @@ public OctoInstallation getInstallation(String name) {
          * @param home the path of the file
          * @return Form validation to present on the Jenkins UI
          */
-        public FormValidation doCheckHome(@QueryParameter String home){
-            if(home != null && !home.isEmpty())
+        public FormValidation doCheckHome(@QueryParameter File home){
+            // this can be used to check the existence of a file on the server, so needs to be protected
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+
+            if(home != null && !home.getPath().isEmpty())
```

Comment:
Confirming: we intend to allow null values for `home` but not empty values? Is our behaviour defined when there is no path to `octo` specified?

Reply:
TBH I don't think Jenkins will even call this method with a null value (https://github.com/jenkinsci/jenkins/blob/449c5aced523a6e66fe3d6a804e5dbfd5c5c67c6/core/src/main/java/hudson/tools/ToolDescriptor.java#L159 doesn't check for null), but the null check was added to be defensive.

Reply:
Never mind - I had a look through the rest of the code properly. We're relying on having `octo` / `Octo.exe` in the path if one isn't specified.

Reply:
(Meh. Human race condition :) )

There's a philosophical discussion to be had about null checks and returning versus throwing, but I'm happy to leave that for another day if you are :)